CDCgov/prime-reportstream
# Moves Services fetch to Session level

This PR moves the Receivers call out of the Daily Data page, re-labels the attribute from "senders" to "services" as receivers are another type of service, and fetches whenever a `MemberType.SENDER` or `MemberType.RECEIVER` loads their session. 

Test Steps:
1. Ensure you have [React Dev Tools](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi?hl=en) installed for your browser (the given link is for the Chrome extension)
2. Use the Components tab in your dev console to watch the state within SessionContext
3. Log in and arrive at the admin Org List. Assert there are **not** any calls labeled "senders" or "receivers"
4. Swap to any org; Assert there are **not** any calls labeled "senders" or "receivers"
5. Swap user type in the "mimic user type" dropdown to Sender; Assert there is a network call labeled "senders" and that the SessionContext membership settings reflect the result in `allServices` and `service` (active service name)
6. Step 5 again, but for Receivers
7. Navigate to Daily Data and ensure the receiver's data can load and that no 400-499s occur.

## Changes
- Banner above Admin Org List to indicate changes to session mocking process
- `MembershipSettings.sender` -> `MembershipSettings.service`
- Addition of `MembershipSettings.allServices` array of available services
- API config for services in `services.ts`
- Addition of `UPDATE_SERVICES_LIST` and `SET_ACTIVE_SERVICE` action types for membership settings
- Removes single use and files for `ReceiverHooks`
- And finally, the fetch itself inside of `useOktaMemberships`

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
